pydantic/dataclasses.py

```diff
@@ -197,7 +189,15 @@ def dataclass(
     def wrap(cls: Type[Any]) -> 'DataclassClassOrWrapper':
         import dataclasses
 
-        if is_builtin_dataclass(cls) and _extra_dc_args(_cls) == _extra_dc_args(_cls.__bases__[0]):  # type: ignore
+        should_use_proxy = (
```

Comment:
I reckon I should have added directly the dataclass kwarg `use_proxy` and invite people to write
`PydanticDC = pydantic.dataclasses.dataclass(DC, use_proxy=True)` to keep `DC` untouched.
It would have been explicit for these rare edge cases and would have avoided this kind of (very poor) magic
But I reckon it's too late now for v1 and v2 won't need this at all yay 🎉